### ISSUE TITLE
How to initialize Features and Properties using an external file, not in classpath

### ISSUE THREAD

**Issue by ckurdekar** [ff4j/ff4j#439]
@clun 

Trying to mount runtime external FF4j xml, however realized its working when its static content loaded from resources i.e. class loader path. Trying to run docker container (eventually it would be configured as configmap in Kubernetes and mount to a container) to runtime detect the k8s config map changes and recreate the FF4J bean.

Is there way to avoid loading from class loader /resources path and read from external configuration path at runtime?

**Comment by clun** [comment]
Yes, you can explicitly provide a file name and a FileInputStream. Also you might be insterested in loading Features as YAML not XML

```java
File myConfigMap = new File("/tmp/config.yaml");
FileInputStream fis = new FileInputStream(myConfigMap);
FF4jConfiguration ff4jConfigBean = new YamlParser().parseConfigurationFile(fis);
FF4j ff4j = new FF4j(ff4jConfigBean);
```

to have the `YamlParser` simply add dependency `ff4j-config-yaml`

I do have a question for you. If you edit a `configMap`, k8s does not reload the pods for you. You need some logic at your side, Here is how to this with Spring Cloud. https://medium.com/swlh/kubernetes-configmap-confuguration-and-reload-strategy-9f8a286f3a44 Would be interested if you come up with a nice solution.

**Comment by ckurdekar** [comment]
> Yes, you can explicitly provide a file name and a FileInputStream. Also you might be insterested in loading Features as YAML not XML
> 
> ```java
> File myConfigMap = new File("/tmp/config.yaml");
> FileInputStream fis = new FileInputStream(myConfigMap);
> FF4jConfiguration ff4jConfigBean = new YamlParser().parseConfigurationFile(fis);
> FF4j ff4j = new FF4j(ff4jConfigBean);
> ```
> 
> to have the `YamlParser` simply add dependency `ff4j-config-yaml`
> 
> I do have a question for you. If you edit a `configMap`, k8s does not reload the pods for you. You need some logic at your side, Here is how to this with Spring Cloud. https://medium.com/swlh/kubernetes-configmap-confuguration-and-reload-strategy-9f8a286f3a44 Would be interested if you come up with a nice solution.

can you share some sample YAML configurations?

**Comment by clun** [comment]
here you go.

https://github.com/ff4j/ff4j/blob/master/ff4j-test/src/main/resources/test-ff4j-features.yml

**Comment by ckurdekar** [comment]
FF4j ff4j = new FF4j(ff4jConfigBean); ... seems not available with ff4j-core 1.8.8 version... when it would be released?
though work around is thr to load feature store and property store using ff4jConfigBean and then assign these 2  stores to ff4j instance

**Comment by clun** [comment]
> when it would be released?
It has been released with `1.8.9` today.
